### DISCUSSION TITLE
Access to multiple texture atlases via single skin builder

### DISCUSSION THREAD

**Discussion by Divelix** [libktx/ktx#329] [Q&A] [answered]
Hi, I have ktx skin builder based on default `uiskin.atlas` + some of my styles (colors, `Pixmap` based textures, etc.). Until recently I used it to layout UI + separate textures for gameplay stuff, but now I grouped those gameplay stuff into 1 atlas and now want to access both (`uiskin.atlas` and `gameplay.atlas`) via `Scene2DSkin.defaultSkin` (which will let me not to pass `AssetManager` instance all over the gameplay code). How can I do that? Or even - am I thinking in right direction? What is the proper way to do such things?

P.S. I don't want to put ui and gameplay textures into 1 atlas because I plan to separate its loading process in time to speed up main menu loading on game startup.

**Comment by czyzby** [comment, marked as answer]
`Skin` has an `addRegions` method that allows you to add regions from another `TextureAtlas`. You have to be careful to avoid name clashes in the texture region file names, as this might override existing resources.

Basically, after your UI `Skin` is constructed, you can load your atlas and do this:

```kotlin
val skin: Skin = TODO()
val atlas: TextureAtlas = TODO()

skin.addRegions(atlas)
```

**Reply by Divelix** [reply]
That works, thanks. But, as I understand, it won't allow me to load ui assets to skin first, show menu, and just then load gameplay assets.

**Reply by czyzby** [reply]
@Divelix It will. You can use `addRegions` at any time.

If you need to define new styles after registering new atlas, you can do so with `apply`:

```kotlin
val uiAtlas = TODO()
// Create skin with basic styles:
val skin = skin(atlas = uiAtlas) {
  // Add common UI component styles here.
  button { TODO() }
}

// Go to the loading view, load game assets.

val gameAtlas = TODO("Load atlas")
skin.apply {
  addRegions(gameAtlas)

  // Define game-specific styles here.
  button { TODO() }
}
```

Just for the record, `apply` is not strictly necessary, as all style definitions are just extensions methods of `Skin`, but it will help you organize the styles into building blocks. I might add a new method such as `register` to make it more explicit.

EDIT: The KTX starting from 1.9.13-b1 allows to use `skin.register` instead of `apply`. Both approaches are similar, but `register` is annotated with the DSL annotation, so it provides some extra safety when nesting styles.
